bazarr/subtitles/refiners/anilist.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+# fmt: off
+
+import logging
+import requests
+from collections import namedtuple
+from datetime import timedelta
+
+from app.config import settings
+from subliminal import Episode, region
+
+logger = logging.getLogger(__name__)
+refined_providers = {'jimaku'}
+
+class AniListClient(object):
+    def __init__(self, session=None):
+        self.session = session or requests.Session()
+
+    AnimeInfo = namedtuple('AnimeInfo', ['anime'])
+    
+    @region.cache_on_arguments(expiration_time=timedelta(days=1).total_seconds())
+    def get_series_mappings(self):
+        r = self.session.get(
+            'https://raw.githubusercontent.com/Fribb/anime-lists/master/anime-list-mini.json',
+            timeout=10
+        )
+
+        r.raise_for_status()
+        return r.json()
+
+    def get_series_id(self, candidate_id_name, candidate_id_value):
+        anime_list = self.get_series_mappings()
+        
+        tag_map = {
+            "series_anidb_id": "anidb_id",
+            "series_tvdb_id": "thetvdb_id",
+            "imdb_id": "imdb_id"
+        }
+        mapped_tag = tag_map.get(candidate_id_name, candidate_id_name)        
+        
+        obj = [obj for obj in anime_list if mapped_tag in obj and str(obj[mapped_tag]) == str(candidate_id_value)]
+        logger.debug(f"Based on '{mapped_tag}': '{candidate_id_value}', anime-list matched: {obj}")
+
+        if len(obj) > 0:
+            return obj[0]["anilist_id"]
+        else:
+            logger.debug(f"Could not find corresponding AniList ID with '{mapped_tag}': {candidate_id_value}")
+            return None
+
+def refine_from_anilist(path, video):
+    # Safety checks
+    if isinstance(video, Episode):
+        season = video.season if video.season else 0
+        if season > 1 and not video.series_anidb_id:
+            logger.error(
+                f"Will not refine '{video.series}' as it only has a TVDB ID, but its season ({season}) is higher than 1." +
+                "\nSuch IDs are not unique to each season; To allow refinement for this series, you must enable AniDB integration."
+            )
+            return
+
+    if refined_providers.intersection(settings.general.enabled_providers) and video.anilist_id is None:
+        refine_anilist_ids(video)
+
+def refine_anilist_ids(video):
+    anilist_client = AniListClient()
+    
+    if isinstance(video, Episode):
+        season = video.season if video.season else 0
+        if season > 1:
+            candidate_id_name = "series_anidb_id"
+        else:
+            candidate_id_name = "series_tvdb_id"
+    else:
+        candidate_id_name = "imdb_id"
```

Comment:
I wonder if we should always rely in the anidb even if the season is the 1, is this guaranteed to get the correct entry?

Reply:
I see what you mean, but when getting identical records for a given TVDB ID, it seems that the first entry in the array is always the first season of a show anyway.  
But I agree, only relying on the anidb ID would be more accurate.